hashednode.go

```diff
@@ -31,14 +31,37 @@ import (
 )
```

Comment:
OK, important changes are coming here. This was an important optimization for the replay benchmark.